x-pack/plugins/ml/index.js

```diff
@@ -63,6 +63,24 @@ export const ml = (kibana) => {
         xpackMainPlugin.info.feature(thisPlugin.id).registerLicenseCheckResultsGenerator(checkLicense);
       });
 
+      xpackMainPlugin.registerFeature({
+        id: 'ml',
+        name: 'Machine Learning',
+        icon: 'mlApp',
+        navLinkId: 'ml',
+        privileges: {
+          all: {
+            api: [],
+            app: ['ml'],
+            savedObject: {
+              all: [],
+              read: [],
```

Comment:
`config` isn't implicit at this point, we probably need this for ML also, I haven't given it a close look yet.

Reply:
The [`renderApp` flow](https://github.com/kobelb/kibana/blob/gap/actions-restructure/src/ui/ui_render/ui_render_mixin.js#L145) requires the `config` object, because those values end up injected into the UI. There's a `renderAppWithDefaultConfig` option that can be called, but that's not the typical flow.
So if a user _only_ has access to the ML app, and nothing else, they'll still need the privilege to read the `config` object.

While I think there's value in making this implicit, I think _for now_ we're better off keeping this explicit.

Reply:
Thanks for doing that research @legrego, i'll make sure it's added to all of the features that we have added right now, since they're all "app based features"